code-challenges.test.js

```diff
@@ -18,54 +18,165 @@
+
+const shuffleColors = (array) =>{
+    //filter the array will slice the index of 0
+    var colorsArray = array.slice(1,)
+    //filter the array will slice the index of 0
+//push (Math.floor *Math.random) for index new array to newArray
+    var i = colorsArray.length, j, temp;
+    while(--i >0) {
+        j= Math.floor(Math.random() * (i+1));
+        temp= colorsArray[j];
+        colorsArray[j] = colorsArray[i];
+        colorsArray[i] = temp;
+    }
+
+return colorsArray
+
+}
+
+//
+
 var colors1 = ["purple", "blue", "green", "yellow", "pink"]
 // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
 var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
 
+console.log(shuffleColors(colors1))
+console.log(shuffleColors(colors2))
 
 
 
-// b) Create the function that makes the test pass.
-
-
 
 
 
 // --------------------2) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with expect statements for each of the variables provided.
+//setting up test that will describe when cubedSum gets called 
+//it will take the argument and *** by itself and add each new number in that array to get a total. 
+describe("when sumUp is called", () =>{
+    it("if argument is cubeAndSum1, return 99",() =>{
+        const argument = cubeAndSum1;
+        const actualResults = sumUp(argument)
+        expect(actualResults).toEqual(99)
+    })
+    it("if argument is cubeAndSum2, return 1125",() =>{
+        const argument = cubeAndSum2;
+        const actualResults = sumUp(argument)
+        expect(actualResults).toEqual(1125)
+    })
+})
+
+
+// b) Create the function that makes the test pass.
+//setting up function that takes in an array of numbers as the argument
+
+
+const sumUp = (array) => {
+    let cubedArray =[]
+    for(let i= 0; i < array.length ; i++){
+        cubedArray.push(array[i] * array[i] * array[i])
+    }
+    return cubedArray.reduce((a,b) => (a + b))
+    
+}
+
+
 
 var cubeAndSum1 = [2, 3, 4]
 // Expected output: 99
 var cubeAndSum2 = [0, 5, 10]
 // Expected output: 1125
 
+console.log(sumUp(cubeAndSum1))
+console.log(sumUp(cubeAndSum2))
 
 
 
 
-// b) Create the function that makes the test pass.
-
 
 
 
 
 // --------------------3) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
 // a) Create a test with expect statements for each of the variables provided.
+//setting up test that will describe when lowestAndBiggest gets called 
+//it will take in the arrgument of an array
+//return index 0 and the last index
+//for argument nums1 expect out to be [-8, 90]
+//for argument nums2 expect out to be [-59, 109]
+
+describe("when sortNums is called", () =>{
+    it("if argument is nums1, return -8, 90",() =>{
```

Comment:
tests are for documenting behavior so test descriptions should describe behavior rather than expected inputs and outputs